kythe/typescript/indexer.ts

```diff
@@ -363,16 +374,43 @@ class Vistor {
   }
 
   /** getSymbolName computes the VName (and signature) of a ts.Symbol. */
-  getSymbolName(sym: ts.Symbol, ns: TSNamespace): VName {
+  getSymbolName(sym: ts.Symbol, ns: TSNamespace, context?: ts.Node): VName {
     let vnames = this.symbolNames.get(sym);
-    if (vnames && vnames[ns]) return vnames[ns]!;
+    let declarations = sym.declarations;
+    if (!(context && declarations.length > 1) && vnames && vnames[ns])
+      return vnames[ns]!;
+    // TODO: update symbolNames table to account for context kind
```

Comment:
I didn't follow this TODO, can you elaborate?

Reply:
The VName for a symbol under some context is not currently cached in symbolNames. We may want to add a field in the map that checks for a context and returns the VName for a symbol with that context later on.